tools/manifest/sourcefile.py

```diff
@@ -993,8 +997,8 @@ def manifest_items(self):
         if drop_cached and "__cached_properties__" in self.__dict__:
             cached_properties = self.__dict__["__cached_properties__"]
             for key in cached_properties:
-                if key in self.__dict__:
-                    del self.__dict__[key]
+                if str(key) in self.__dict__:
```

Comment:
This is the only bit in the PR that looks a bit unsure about what the types are and looks like defensive coding. Can we instead ensure that the keys are strings in utils.py, which appears to be the only place where things are added to this dict? There, it looks to me like it is already guaranteed to be a str, as the input comes from `func.__name__`.

Reply:
So I don't really understand what broke here. mypy was grumpy about the types until I made this change, but afaict it's a noop from the point of view of the code since the dictionary keys are already the `str` type in all cases. I also can't see why this PR should have changed things; maybe it just needs a type annotation somewhere else or I got unlucky with a mypy change? I'm hoping @gsnedders has some insight.

Reply:
For the record; I did some digging, it's because we're reusing a variable name and mypy gets confused (https://github.com/web-platform-tests/wpt/pull/23669#issuecomment-630381158). We'll either rename it or explicitly type it.